### ISSUE TITLE
Rename Just / Nothing to Some / None

### ISSUE THREAD

**Issue by Pauan** [koka-lang/koka#39]
I never liked Haskell's naming convention for the `Maybe` type.

I propose to rename the constructors to `Some` and `None`:

```
type maybe<a> {
  con None
  con Some( value : a )
}
```

This has some advantages:

* Shorter.

* They have the same number of characters (4), so it looks better:

   ```
   match (a) {
     Just(value) -> Just(value)
     Nothing     -> Nothing
   }

   match (a) {
     Some(value) -> Some(value)
     None        -> None
   }
   ```

   ```
   fun foo() {
     if (bar)
     then Just(1)
     else Nothing
   }

   fun foo() {
     if (bar)
     then Some(1)
     else None
   }
   ```

* `Just` sounds very strange to me. If read out loud, `Just(a)` means that it's "just an `a`", which doesn't make sense. `Some` sounds more natural to me, since it's a short form of "something"

* Rust / OCaml / F# all use `Some` / `None`

Disadvantages:

* It looks worse when pattern matching with a single-character variable (in this case `b`):

   ```
   match (a) {
     Just(b) -> Just(b)
     Nothing -> Nothing
   }

   match (a) {
     Some(b) -> Some(b)
     None    -> None
   }
   ```

* Haskell / PureScript / Elm all use `Just` / `Nothing`

**Comment by rauchg** [comment]
Strongly agree!

**Comment by timbertson** [comment]
I just used `foo.is-just` and it's another good argument for this rename - `foo.is-some` makes much more sense

**Comment by OniHil** [comment]
I agree with the naming, and wouldn't mind making the change. However I'm concerned about potential confusion since ["some" is reserved](https://github.com/koka-lang/koka/blob/d3b244c7783b3a97670974a202f07937ccacf27a/src/Syntax/Lexer.x#L303-L309). As for it being used, it's in the [v1/std async package](https://github.com/koka-lang/koka/blob/d3b244c7783b3a97670974a202f07937ccacf27a/lib/v1/std/async.kk#L414-L448). It's also getting added to std with #852 

Here's an example of how it could look:

```koka
// Renamed definition
pub value type maybe<a>
  con None
  con Some( value : a )

fun example( x : a ) : maybe<a> {
  var temp : some<a> maybe<a> := Some(x)
  temp
}
```

It is a significant change that would break things, and the example might be a slight edge case. Perhaps it is still worth changing?
